storage/ndb/rest-server2/server/src/main.cc

```diff
@@ -293,17 +322,28 @@ int main(int argc, char *argv[]) {
   if (globalConfigs.rondis.enable) {
     g_rondis_conn_factory = new RondisConnFactory();
     g_rondis_handle = new RondisHandle();
-    // todo use globalConfigs.rondis.databases to configure individual databases.
-    // todo let rondis use g_rondbConnection
-    // todo if thread exits on its own accord (without StopThread() called in
-    // do_exit), make sure to set g_rondis_running = false and then call
-    // do_exit().
-    g_rondis_thread = NewDispatchThread(globalConfigs.rondis.serverPort,
+    g_database_index = (int*)malloc(sizeof(int) * num_rondis_threads);
+    memset(g_database_index, 0, sizeof(int) * num_rondis_threads);
+    bool dirty_incr_decr_flag[MAX_NUM_DATABASES];
+    memset(&dirty_incr_decr_flag[0], 0, sizeof(dirty_incr_decr_flag));
+    for (Uint32 i = 0; i < globalConfigs.rondis.numDatabases; i++) {
+      dirty_incr_decr_flag[i] = globalConfigs.rondis.databases[i].dirtyIncrDecr;
+    }
+    g_rondis_thread = NewDispatchThread(globalConfigs.rondis.serverIP,
+                                        globalConfigs.rondis.serverPort,
                                         globalConfigs.rondis.numThreads,
                                         g_rondis_conn_factory,
                                         1000,
                                         1000,
                                         g_rondis_handle);
+    setup_ndb_connection_for_rondis(get_ndb_object,
+                                    return_ndb_object,
+                                    exit_on_rondis_error,
+                                    num_rdrs_threads,
```

Comment:
Are you sure? Or should it be `num_rdrs_threads + num_purge_threads`?

Reply:
I am sure, first comes the RDRS threads, second the Rondis threads and last the purge threads